test/e2e/node/pods.go

```diff
@@ -415,6 +415,166 @@ var _ = SIGDescribe("Pods Extended", func() {
+			pod.Spec.InitContainers = []v1.Container{{
+				Name:  "init-container",
+				Image: imageutils.GetE2EImage(imageutils.BusyBox),
+			}}
+
+			ginkgo.By("submitting the pod to kubernetes")
+			pod = podClient.CreateSync(ctx, pod)
+			ginkgo.DeferCleanup(func(ctx context.Context) error {
+				ginkgo.By("deleting the pod")
+				return podClient.Delete(ctx, pod.Name, metav1.DeleteOptions{})
+			})
+
+			ginkgo.By("setting up selector")
+			selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": value}))
+			options := metav1.ListOptions{LabelSelector: selector.String()}
+
+			ginkgo.By("verifying pod generation bumps as expected")
+			tests := []struct {
+				name                 string
+				updateFn             func(*v1.Pod)
+				expectGenerationBump bool
+			}{
+				{
+					name:                 "verifying the new pod's generation is 1",
+					updateFn:             func(pod *v1.Pod) {},
+					expectGenerationBump: false,
+				},
+
+				{
+					name: "updating Tolerations to trigger generation bump",
+					updateFn: func(pod *v1.Pod) {
+						pod.Spec.Tolerations = []v1.Toleration{
+							{
+								Key:      "foo-" + string(uuid.NewUUID()),
+								Operator: v1.TolerationOpEqual,
+								Value:    "bar",
+								Effect:   v1.TaintEffectNoSchedule,
+							},
+						}
+					},
+					expectGenerationBump: true,
+				},
+
+				{
+					name: "updating ActiveDeadlineSeconds to trigger generation bump",
+					updateFn: func(pod *v1.Pod) {
+						int5000 := int64(5000)
+						pod.Spec.ActiveDeadlineSeconds = &int5000
+					},
+					expectGenerationBump: true,
+				},
+
+				{
+					name: "updating container image to trigger generation bump",
+					updateFn: func(pod *v1.Pod) {
+						pod.Spec.Containers[0].Image = imageutils.GetE2EImage(imageutils.Nginx)
+					},
+					expectGenerationBump: true,
+				},
+
+				{
+					name: "updating initContainer image to trigger generation bump",
+					updateFn: func(pod *v1.Pod) {
+						pod.Spec.InitContainers[0].Image = imageutils.GetE2EImage(imageutils.Pause)
+					},
+					expectGenerationBump: true,
+				},
+
+				{
+					name: "updates to pod metadata should not trigger generation bump",
+					updateFn: func(pod *v1.Pod) {
+						pod.SetAnnotations(map[string]string{"key": "value"})
+					},
+					expectGenerationBump: false,
+				},
+
+				{
+					name: "pod generation updated by client should be ignored",
+					updateFn: func(pod *v1.Pod) {
+						pod.SetGeneration(1)
+					},
+					expectGenerationBump: false,
+				},
+			}
+
+			expectedPodGeneration := 1
+			for _, test := range tests {
+				ginkgo.By(test.name)
+				podClient.Update(ctx, podName, test.updateFn)
+				pods, err := podClient.List(ctx, options)
+				framework.ExpectNoError(err, "failed to query for pod")
+				gomega.Expect(pods.Items).To(gomega.HaveLen(1))
+				if test.expectGenerationBump {
+					expectedPodGeneration++
+				}
+				gomega.Expect(pods.Items[0].Generation).To(gomega.BeEquivalentTo(expectedPodGeneration))
```

Comment:
Does a failure here abort the test, or keep going? If it keeps going, the expected generation will be off and subsequent errors will be misleading. I think it aborts, but would be good to confirm.

Reply:
I confirmed that it aborts, so we don't need to worry about potential subsequent errors